__tests__/app.test.js

```diff
@@ -395,3 +394,151 @@ describe("PATCH /api/articles/:article_id --- error handling", () => {
 
 
 
+
+
+
+// ======== 07 ==========
+
+describe("POST /api/articles/:article_id/comments", () => {
+  const validInput = { username: "lurker", body: "a valid body" };
+
+  test("200 status code received when calling api correctly", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(validInput)
+      .expect(200);
+  });
+
+  test("returns a single comment object in the format {comment: [{comment-obj}]}", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(validInput)
+      .expect(200)
+      .then((data) => {
+        expect(data.body).toMatchObject({
+          comment: expect.any(Object),
+        });
+        expect(Array.isArray(data.body.comment)).toBe(true);
+        expect(data.body.comment.length).toBe(1);
+      });
+  });
+
+  test("comment returned is valid format", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(validInput)
+      .expect(200)
+      .then((data) => {
+        expect(data.body.comment[0]).toMatchObject({
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+
```

Comment:
Please can we change these tests to show a 201 status when posting a new comment.

Reply:
Done - now get 201 for all successful POSTs